app/src/backend/HiapService.ts

```diff
@@ -0,0 +1,466 @@
+import { LANGUAGES, ACTION_TYPES } from "@/util/types";
+import {S3Client } from "@aws-sdk/client-s3";
+import { Readable } from "stream";
+import { logger } from "@/services/logger";
+import { db } from "@/models";
+import PopulationService from "./PopulationService";
+import {
+  getTotalEmissionsBySector,
+  EmissionsBySector,
+} from "./ResultsService";
+import {
+  HighImpactActionRanking,
+} from "@/models/HighImpactActionRanking";
+import { HighImpactActionRankingStatus } from "@/util/types";
+import {
+  startPrioritization,
+  checkPrioritizationProgress,
+  getPrioritizationResult,
+} from "./HiapApiService";
+import { InventoryService } from "./InventoryService";
+import GlobalAPIService from "./GlobalAPIService";
+
+export interface CityContextData {
+  locode: string;
+  populationSize: number | null;
+}
+
+export interface CityEmissionsData {
+  stationaryEnergyEmissions: number | null;
+  transportationEmissions: number | null;
+  wasteEmissions: number | null;
+  ippuEmissions: number | null;
+  afoluEmissions: number | null;
+}
+
+export interface PrioritizerCityData {
+  cityContextData: CityContextData;
+  cityEmissionsData: CityEmissionsData;
+}
+
+export interface PrioritizerRequest {
+  cityData: PrioritizerCityData;
+}
+
+export interface PrioritizerResponseMetadata {
+  locode: string;
+  rankedDate: string;
+}
+
+export interface PrioritizerRankedAction {
+  actionId: string;
+  rank: number;
+  explanation: {
+    en: string;
+    es: string;
+    pt: string;
+  };
+}
+
+export interface PrioritizerResponse {
+  metadata: PrioritizerResponseMetadata;
+  rankedActionsMitigation: PrioritizerRankedAction[];
+  rankedActionsAdaptation: PrioritizerRankedAction[];
+}
```

Comment:
It might make sense to extract these type definitions (and the ones below) to a separate file?